### DISCUSSION TITLE
Does pnpm still generate / use / recommend a "shrinkwrap.yaml" file?

### DISCUSSION THREAD

**Discussion by openjck** [pnpm/pnpm#8595] [Q&A] [answered]
NPM is capable of generating two different kinds of lockfiles: _package-lock.json_ and _npm-shrinkwrap.json_. To my surprise, _package-lock.json_ is not simply a newer version of _npm-shrinkwrap.json_; the two files serve different purposes, and [there are good reasons](https://stackoverflow.com/a/46132512/715866) to use one rather than the other. Personally, I find it pretty confusing, but that's how NPM works.

I know that pnpm generates a _pnpm-lock.yaml_ file. I tried researching whether pnpm can also generate shrinkwrap files, perhaps because they're useful in different situations, like in NPM. I've found some old GitHub issues that mention a _shrinkwrap.yaml_ file, and even one occurrence in [what appears to be official documentation](https://pnpm.io/community/tools#tools-with-pnpm-support:~:text=lock.json%20or-,shrinkwrap.yaml,-changes%20due%20to). Aside from those instances, it appears _shrinkwrap.yaml_ is undocumented, leading me to believe it's not generated, used, or recommended any more.

Is _shrinkwrap.yaml_ still generated? Should it be used? Should it be checked into source control? Should it be used in certain situations, like _npm-shrinkwrap.json_ should?

**Comment by zkochan** [comment]
no, pnpm never generated a `npm-shrinkwrap.json`. In my opinion the whole idea of `npm-shrinkwrap.json` is not good because you have to basically promise to forever support this "lockfile" format that ships with the dependency. If you want to lock the dependencies of your package, then bundle your package with esbuild or use "bundledDependencies".

**Reply by zkochan** [reply, marked as answer]
`shrinkwrap.yaml` was just the old name of `pnpm-lock.yaml` but it never worked as `npm-shrinkwrap.json`, it worked as a regular lockfile.

**Reply by openjck** [reply]
Thank you very much. In that case, if someone is using pnpm, do you recommend checking in _pnpm-lock.json_ in (almost) all cases, regardless of whether one is building a package or an application?

**Reply by zkochan** [reply]
yes, the lockfile should be committed to the repository.
